Browserstack multiremote fix

Ankit098 opened the pull request:
## Proposed changes

Handle case where bstack:options in caps causes issues with multiremote

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### Reviewers: @webdriverio/project-committers

christian-bromann commented on packages/wdio-browserstack-service/src/util.ts:
```diff
@@ -44,7 +44,7 @@ export function getBrowserCapabilities(browser: Browser<'async'> | MultiRemoteBr
  */
 export function isBrowserstackCapability(cap?: Capabilities.Capabilities) {
-    return Boolean(cap && cap['bstack:options'])
+    return Boolean(cap && cap['bstack:options'] && Object.keys(cap['bstack:options']).length && !(Object.keys(cap['bstack:options']).length === 1 && cap['bstack:options'].wdioService))
```
Can we make this a bit more readable and also add a comment what you are checking exactly here?
```suggestion
    return (
        Boolean(cap && cap['bstack:options'] &&
        !(
            Object.keys(cap['bstack:options']).length === 1 &&
            cap['bstack:options'].wdioService
        )
    )
```

christian-bromann replied:
let's add a comment to the code with the explanation

Ankit098 replied:
Hey, checking further into this issue as found another scenario where this error is thrown (which is independent of the code adding wdioService cap)  - when multiremote is running on Automate + AppAutomate sessions. Will look into the fix and push changes to handle that.

christian-bromann reviewed (commented):
The other option is to __not__ set the `wdioService` bs option if no other options are defined.

christian-bromann commented:
@Ankit098 any updates to this?

Ankit098 commented:
Hey, addressed the concerns mentioned in this PR - regarding readability - this change fixes the trigger usecase (multiremote with one local browser and one browserstack automate browser).. for similar warning in some other use cases, will look into it in next PRs but on a lesser priority as this does not hamper test runs in any way.

christian-bromann reviewed (approved):
👍  Thank you!
